airflow/models.py

```diff
@@ -4519,6 +4519,7 @@ def verify_integrity(self, session=None):
                 logging.info("Restoring task '{}' which was previously removed from DAG '{}'".format(ti, dag))
                 Stats.incr("task_restored_to_dag.{}".format(dag.dag_id), 1, 1)
                 ti.state = State.NONE
+            session.merge(ti)
```

Comment:
yeah, I think you need that to persist back to db.

Reply:
I'm not following - something seems off here. On the line right above this we are also modifying the `ti` instance. Why didn't we need merge() before?

Based on [the docs](https://docs.sqlalchemy.org/en/13/orm/session_state_management.html#merging), I'd expect that we'd only need to `merge()` when we are transferring attributes from some external object into the session, but we fetched this `ti` from the same `session` that we are using here, so I don't see what this changes

Reply:
Yeah weird thing is that this is not required in my local setup (SequentailExecutor+sqlite)

Reply:
I think we need to `merge()` before, Jin also found out that the state change is not persisted:
https://lyft.slack.com/archives/CTKA6PTEV/p1589323151085100